packages/api-server/src/methods/gw-error.ts

```diff
@@ -234,3 +234,29 @@ export function failedReasonByErrorReceipt(
 
   return failedReason;
 }
+
+export function parseGwRunResultError(err: any): RpcError {
+  const gwErr = parseGwError(err);
+  const failedReason: any = {};
+  if (gwErr.statusCode != null) {
+    failedReason.status_code = "0x" + gwErr.statusCode.toString(16);
+    failedReason.status_type = evmcCodeTypeMapping[gwErr.statusCode.toString()];
+  }
+  if (gwErr.statusReason != null) {
+    failedReason.message = gwErr.statusReason;
+  }
+  let errorData: any = undefined;
+  if (Object.keys(failedReason).length !== 0) {
+    errorData = { failed_reason: failedReason };
+  }
+
+  let errorMessage = gwErr.message;
+  if (gwErr.statusReason != null && failedReason.status_type != null) {
```

Comment:
When `evmcCodeTypeMapping` doesn't contains the `gwErr.statusCode`, the `failedReason.status_type` will be assigned to `undefined` but not `null`.

Reply:
Should we use `if (!gwErr.statusReason && !failedReason.status_type)` instead?

Reply:
!= null will check undefined and null

Reply:
Yeah, you are right. Magic juggling-check 😂